src/apps/filtering.js

```diff
@@ -1,10 +1,124 @@
 // @flow
+import invariant from "invariant";
 import type { App } from "../types/manager";
+import type { InstalledItem } from "./types";
+import { getCryptoCurrencyById, isCurrencySupported } from "../currencies";
 
-export type SortOptions = *;
+export type SortOptions = {
+  type: "name" | "marketcap",
+  order: "asc" | "desc",
+  marketcapTickers?: string[]
+};
 
-export const sortApps = (apps: App[], _options: SortOptions): App[] => apps;
+export type FilterOptions = {
+  query?: string,
+  installedApps?: InstalledItem[],
+  type: "all" | "installed" | "not_installed" | "supported" | "updatable"
+};
 
-export type FilterOptions = *;
+const missingInstalledApps =
+  "To filter by installed/updatable apps you need to provide the installed list from state.";
 
-export const filterApps = (apps: App[], _options: FilterOptions): App[] => apps;
+const missingMarketcapTickers =
+  "In order to sort by marketcap you need to provide the marketcap tickers.";
+
+const missingCurrencyId =
+  "Although we are already filtering the apps to currency apps, this is needed for flow.";
+
+const searchFilter = (query?: string) => ({ name, currencyId }) => {
+  if (!query) return true;
+  const currency = currencyId ? getCryptoCurrencyById(currencyId) : null;
+  const terms = `${name} ${
+    currency ? `${currency.name} ${currency.ticker}` : ""
+  }`;
+  return terms.toLowerCase().includes(query.toLowerCase().trim());
+};
+
+const typeFilter = (type = "all", installedApps?: InstalledItem[]) => app => {
+  switch (type) {
+    case "installed":
+      invariant(installedApps, missingInstalledApps);
+      return installedApps.find(a => a.name === app.name);
+    case "not_installed":
+      invariant(installedApps, missingInstalledApps);
+      return !installedApps.find(a => a.name === app.name);
+    case "updatable":
+      invariant(installedApps, missingInstalledApps);
+      return installedApps.find(a => a.name === app.name && !a.updated);
```

Comment:
it's better if we have beforehand a map of {installed} so we don't need to .find() at each app iteration